File: Programmers/LV.1/이상한 문자 만들기.py
# def solution(s):
#     answer = ''
#     txt = s.split(' ')
#     for i in range(len(txt)):
#         for j in range(len(txt[i])):
#             if j % 2:  # 홀수 idx
#                 answer += txt[i][j].lower()
#             else:  # 짝수 idx
#                 answer += txt[i][j].upper()
#         answer += ' '
#     return answer[:-1]


####
def solution(s):
    s_list = list(s.title())
    tmp = -1
    for i in range(len(s_list)):
        if s_list[i].isupper():
            tmp = i
        elif not (i-tmp) % 2: # 짝수일때
            s_list[i] = s[i].upper()
    return ''.join(s_list)


print(solution('try hell world'))

File: Programmers/LV.1/이상한문자만들기.js
const s = "try hello world";
// 짝수번째는 대문자, 홀수번째는 소문자로 바꿔 리턴

function solution(s) {
  var answer = "";
  const word = s.split(" ");

  for (let i = 0; i < word.length; i++) {
    for (let j in word[i]) {
      answer =
        j % 2
          ? (answer += word[i][j].toLowerCase())
          : (answer += word[i][j].toUpperCase());
    }
    answer += " ";
  }
  console.log(answer);
  return answer.slice(0, -1);
}

solution(s);
